lib/rules/no-magic-numbers.js

```diff
@@ -26,10 +26,129 @@ function normalizeIgnoreValue(x) {
 	return x;
 }
 
+/**
+ * Gets the true parent of the literal, handling prefixed numbers (-1 / +1)
+ * @param {ASTNode} node The literal node
+ * @returns {ASTNode} The parent node
+ */
+function getLiteralParent(node) {
+	if (["-", "+"].includes(node.parent.operator)) {
+		return node.parent.parent;
+	}
+
+	return node.parent;
+}
+
+/**
+ * Checks if the node parent is a TypeScript enum member
+ * @param {ASTNode} node The node to be validated
+ * @returns {boolean} True if the node parent is a TypeScript enum member
+ */
+function isParentTSEnumDeclaration(node) {
+	const parent = getLiteralParent(node);
+	return parent.type === "TSEnumMember";
+}
+
+/**
+ * Checks if any ancestor of the node is a TypeScript type alias declaration
+ * @param {ASTNode} node The node to be validated
+ * @returns {boolean} True if any ancestor is a TypeScript type alias declaration
+ */
+function isInsideTSTypeAliasDeclaration(node) {
+	let current = node;
+
+	while (current && current.parent) {
+		if (current.parent.type === "TSTypeAliasDeclaration") {
+			return true;
+		}
+		current = current.parent;
+	}
+
+	return false;
+}
```

Comment:
If we allow numbers anywhere inside type alias declarations, that would differ from how `@typescript-eslint/no-magic-numbers` works. For example, it doesn't allow numbers in object types:

```ts
/* eslint @typescript-eslint/no-magic-numbers: ["error", { "ignoreNumericLiteralTypes": true }] */

type Foo = { bar: 42 }; // error
```

Reply:
Good catch, fixed.